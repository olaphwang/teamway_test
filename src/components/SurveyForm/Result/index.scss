.result {
  text-align: center;
}

.result-header {
  position: relative;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 3em;
  line-height: 41%;
  letter-spacing: 0.2em;
  color: #000;
}

.title {
  margin: 20px 20px 0;
}

.description {
  margin: 10px 20px 0;
  line-height: 130%;
}
