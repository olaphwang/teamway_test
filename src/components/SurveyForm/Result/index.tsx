import React from 'react';
import { SelectedAnswers } from '@stypes/questions';
import styles from './index.scss';

interface ResultProps {
  selectedAnswers: SelectedAnswers;
}

const Result = ({ selectedAnswers }: ResultProps) => {
  const score = Object.values(selectedAnswers).reduce((total, v) => (total += v.value), 0);
  return (
    <div className={styles.result}>
      <div className={styles['result-header']}>
        <strong>{score}</strong>
      </div>

      <div className={styles.title}>Your personality trait</div>

      <div className={styles.description}>Your score is {score}</div>
    </div>
  );
};

export default Result;
