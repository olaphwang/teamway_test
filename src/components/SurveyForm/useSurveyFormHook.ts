import { useState, useCallback, useEffect } from 'react';
import { Questions, SelectedAnswers } from '@stypes/questions';
import questions from '@mockData/questions';

const useSurveyFormHook = () => {
  const [questionIndex, setquestionIndex] = useState<number>(-1);
  const [selectedAnswers, setselectedAnswers] = useState<SelectedAnswers>({});
  const [quetionsData, setquestionsData] = useState<Questions[]>([]);

  useEffect(() => {
    setquestionsData(questions.map((q) => ({ ...q, answers: q.answers.sort(() => 0.5 - Math.random()) })));
  }, []);

  const handleNext = useCallback(() => {
    if (questionIndex !== quetionsData.length) setquestionIndex((questionIndex) => questionIndex + 1);
  }, [questionIndex, quetionsData]);

  const handlePrev = useCallback(() => {
    if (questionIndex !== 0) setquestionIndex((questionIndex) => questionIndex - 1);
  }, [questionIndex, quetionsData]);

  return {
    questionIndex,
    setquestionIndex,
    selectedAnswers,
    setselectedAnswers,
    quetionsData,
    setquestionsData,
    handleNext,
    handlePrev,
  };
};

export default useSurveyFormHook;
