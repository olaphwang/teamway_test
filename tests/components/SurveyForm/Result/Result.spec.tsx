import React from 'react';
import renderer from 'react-test-renderer';
import Result from '@components/SurveyForm/Result';
import { renderHook } from '@testing-library/react';
import useSurveyFormHook from '@components/SurveyForm/useSurveyFormHook';

describe('Result component', () => {
  test('should render component properly', () => {
    // when
    const {
      result: { current },
    } = renderHook(() => useSurveyFormHook());
    const { selectedAnswers } = current;
    const componentRenderer = renderer.create(<Result selectedAnswers={selectedAnswers} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
