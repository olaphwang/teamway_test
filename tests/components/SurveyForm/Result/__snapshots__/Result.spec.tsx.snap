// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component should render component properly 1`] = `
<div
  className="result"
>
  <div
    className="result-header"
  >
    <strong>
      0
    </strong>
  </div>
  <div
    className="title"
  >
    Your personality trait
  </div>
  <div
    className="description"
  >
    Your score is 
    0
  </div>
</div>
`;
